pyproject.toml

```diff
@@ -144,13 +148,88 @@ testpaths = [
 [tool.ruff]
 include = ["prymer/**/*.py", "tests/**/*.py"]
 line-length    = 100
-target-version = "py311"
+target-version = "py312"
 output-format  = "full"
 
 [tool.ruff.lint]
-select    = ["C901", "B", "E", "F", "I", "W", "Q"]
-ignore    = ["E203", "E701"]
+select    = [
+    "ARG",    # Unused arguments
+    "C901",   # McCabe complexity
+    "B",      # bugbear
+    "D",      # pydocstyle (docstrings. We have the "google" convention enabled)
+    "D204",   # Blank line between class docstring and first (__init__) method
+    "D213",   # Summary line should be located on the line after opening quotes
+    "E",      # pycodestyle errors
+    "LOG",    # flake8-logging
+    "LOG015", # (preview rule) Prohibit calls to the root logger
+    "F",      # pyflakes
+    "I",      # isort
+    "N",      # PEP8 naming
+    "W",      # pycodestyle warnings
+    "Q",      # flake8-quotes
+]
+ignore    = [
+    "E203",
+    "E701",
+    "D212",  # summary line should be located on the same line as opening quotes
+    "D100",  # missing docstring in public module
+    "D104",  # missing docstring in public package
+    # TODO: fixup the documentation so we can remove the following lints from the ignore list:
+    "ARG001", # unused function argument
+    "ARG003", # unused class method argument
+    "D102",  # missing docstring in public method
+    "D103",  # missing docstring in public function
+    "D105",  # missing docstring in magic method
+    "D107",  # missing docstring in __init__
+    "D214",  # section is over-indented
+    "D200",  # one-line docstring should fit on one line
+    "D202",  # no blank lines allowed after function docstring
+    "D205",  # 1 blank line required between summary line and description
+    "D209",  # multi-line docstring closing quotes should be on a separate line
+    "D213",  # multi-line docstring summary should start at the second line
+    "D412",  # no blank lines allowed between a section header and its content ("Example")
+    "D414",  # section has no content
+    "D415",  # first line should end with a period, question mark, or exclamation point
+    "D417",  # missing argument description in the docstring
+    "F841",  # local variable `x` is assigned to but never used
+    "N815",  # variable `x` in class scope should not be mixedCase
```

Comment:
I added in our ruff lints but then ignored all the ones that failed.

Fixing these will make the API docs render even more nicely.